aea/protocols/base.py

```diff
@@ -165,12 +167,17 @@ def __str__(self):
             + ")"
         )
 
+    def encode(self) -> bytes:
+        """Encode the message."""
+        return self.serializer.encode(self)
+
 
 class Encoder(ABC):
     """Encoder interface."""
 
+    @staticmethod
```

Comment:
these are static methods; fixed now

Reply:
the reason why they were non-static is that the serializer developer might want to be stateful on the serialization side, for some reason. Very rare use case though.

Reply:
Well they cannot now. It seemed like an obvious design flaw before and we do not support statefulness of the Serializer framework side anyway. We were creating a new instance of the Serializer on each occasion; very strange!

Reply:
"Statefulness" means they could have had an initializer with custom attributes that use in the serialization. Now this is not possible at all, since the `staticmethod`. Probably `classmethod` would be a good compromise.

Reply:
It also means a much more complicated design. Previously we had a separate instance of serializer for each message; that is not optimal by any means. I still don't see what the usecase would be where the serializer is stateful and in particular why we need an initializer. All messages of a given protocol should be serialized identially, so I think the class is the right level; not the instance.

Reply:
With classmethod you don't need the initializer.